app/models/patient_import_row.rb

```diff
@@ -222,19 +168,107 @@ def nhs_number_value
 
   private
 
+  def build_patient_import_attributes
+    {
+      address_line_1: address_line_1&.to_s,
+      address_line_2: address_line_2&.to_s,
+      address_postcode: address_postcode&.to_postcode,
+      address_town: address_town&.to_s,
+      birth_academic_year: birth_academic_year_value,
+      date_of_birth: date_of_birth.to_date,
+      family_name: last_name.to_s,
+      gender_code: gender_code_value,
+      given_name: first_name.to_s,
+      nhs_number: nhs_number_value,
+      preferred_family_name: preferred_last_name&.to_s,
+      preferred_given_name: preferred_first_name&.to_s,
+      registration: registration&.to_s
+    }.compact
+  end
+
+  def prepare_patient_changes(patient, import_attributes)
+    patient.registration =
+      import_attributes.delete(:registration) unless stage_registration?
+
+    auto_accept_attributes_if_applicable(patient, import_attributes)
+    handle_address_updates(patient, import_attributes)
+    stage_and_handle_pending_changes(patient, import_attributes)
+
+    patient
+  end
+
+  def handle_address_updates(patient, import_attributes)
+    if address_postcode.present? &&
+         address_postcode.to_postcode != patient.address_postcode
+      import_attributes.merge!(
+        address_line_1: address_line_1&.to_s,
+        address_line_2: address_line_2&.to_s,
+        address_town: address_town&.to_s
+      )
+    elsif auto_overwrite_address?(patient)
+      patient.address_line_1 = import_attributes.delete(:address_line_1)
+      patient.address_line_2 = import_attributes.delete(:address_line_2)
+      patient.address_town = import_attributes.delete(:address_town)
+    end
+  end
+
+  def stage_and_handle_pending_changes(patient, import_attributes)
+    auto_accepted_changes = patient.changes.except("pending_changes")
+
+    patient.stage_changes(import_attributes)
+
+    # If there are pending changes that require review, we need to revert
+    # any auto-accepted changes and move them to pending_changes instead.
+    # This ensures all changes are reviewed together rather than having
+    # some changes applied immediately while others await review. This
+    # is particularly critical when handling potential duplicates like twins,
+    # where auto-accepting some changes could lead to data from one twin being
+    # incorrectly applied to another twin's record.
+    if patient.pending_changes.any?
+      auto_accepted_changes.each do |attribute, (old_value, new_value)|
+        patient[attribute] = old_value
+        patient.pending_changes[attribute] = new_value
+      end
```

Comment:
```suggestion
      patient.restore_attributes(auto_accepted_changes)
      patient.pending_changes.merge!(auto_accepted_changes)
```

Reply:
Turns out this didn't work. We need to make some adjustments given `patient.changes` returns something like `{"given_name" => ["Fabiola", "Pete"]}`. I'm leaning towards the loop now, but happy either way. 😄 

```
new_values = auto_accepted_changes.transform_values { |values| values.last }
patient.restore_attributes(auto_accepted_changes.keys)
patient.pending_changes.merge!(new_values)
```